pkg/services/ngalert/state/state.go

```diff
@@ -174,12 +174,17 @@ func (a *State) resultNoData(alertRule *models.AlertRule, result eval.Result) {
 }
 
 func (a *State) NeedsSending(resendDelay time.Duration) bool {
-	if a.State == eval.Pending || a.State == eval.Normal && !a.Resolved {
+	if a.State == eval.Pending {
```

Comment:
Style nit: what about seeting a boolean result and returning it at the end of the conditional instead of doing a return in each branch?

Reply:
Could you give an example?

Reply:
Even saying you go with a switch, the general pattern I'm thinking of is:

```go
outcome := defaultValue
switch cond {
case "A":
  outcome = foo()
case "B":
  outcome = bar()
}
return outcome
```

Now, some tools get made about the missing `default:`, in which case:
```go
var outcome typ
switch cond {
case "A":
  outcome = foo()
case "B":
  outcome = bar()
default:
  outcome = defaultValue
}
return outcome
```

The thing I like is that you never forget to return. I suppose the downside is that you can forget to correctly assign `outcome` and compiler won't complain because of missing returns.

Like I said, total nit, do what feels best to you.